Lib/_pyio.py

```diff
@@ -978,6 +978,16 @@ def tell(self):
             raise ValueError("tell on closed file")
         return self._pos
 
+    def peek(self, size=1):
+        if self.closed:
+            raise ValueError("peek on closed file")
+        # Due to slicing semantics, this works correctly
+        # even if the size is greater than the buffer length or
+        # the position is beyond the end of the buffer
+        if size < 1:
+            size = len(self._buffer) - self._pos
```

Comment:
This code looks kind of complicated, whereas you can just do:

```py
if size < 1:
    return self._buffer[self._pos:]
```

Reply:
Changed